src/Bedrock.Framework.Experimental/Protocols/Amqp/Methods/ConnectionStart.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Bedrock.Framework.Experimental.Protocols.Amqp.Methods
+{
+    public class ConnectionStart : MethodBase, IAmqpMessage
+    {
+        public override byte ClassId => 10;
+        public override byte MethodId => 10;
+
+        public byte VersionMajor { get; private set; }
+        public byte VersionMinor { get; private set; }
+        public Dictionary<string, object> ServerProperties { get; private set; }
+        public ReadOnlyMemory<byte> SecurityMechanims { get; private set; }
+        public ReadOnlyMemory<byte> Locale { get; private set; }
+
+        public bool TryParse(ReadOnlySequence<byte> input,  out SequencePosition end)
+        {
+            SequenceReader<byte> reader = new SequenceReader<byte>(input);
+
+            reader.TryRead(out var verionMajor);
+            reader.TryRead(out var versionMinor);
+
+            VersionMajor = verionMajor;
+            VersionMinor = versionMinor;           
+            try
+            {
+                ServerProperties = ProtocolHelper.ReadTable(ref reader);
+                var security = ProtocolHelper.ReadLongString(ref reader);
+                if(security.Span.IndexOf(Encoding.UTF8.GetBytes("PLAIN AMQPLAIN").AsSpan()) >= 0)
```

Comment:
For `Encoding.UTF8.GetBytes("PLAIN AMQPLAIN"` you can store the bytes as ROS (like you have done in other classes), so no need to get the ASCII-bytes every time the method is called.

```c#
ReadOnlySpan<byte> plainAMQPlain = new byte [] { (byte)'P', ... };
```
I know it's a pity to type it that way, but there are issues out (at Roslyn) to improve this.

Alternatively you can use a field:
```c#
private static readonly byte[] s_amqPlain = Encoding.UTF8.GetBytes("...");
```

Reply:
why static ? private readonly ok, but i'm not following you for static

Reply:
The JIT treats `static readonly` as kind of a constant. I.e. it can produce better code at runtime.
As the value never changes, it can be the same for all instances, so `static`.

Reply:
I did'nt know. When you say it can produce better code ta runtime, when use static readonly vs constant ?

Reply:
> when use static readonly vs constant ?

If it's like a constant, but you can't use `const`, then use `static readonly`.

`Encoding.UTF8.GetBytes("...")` can't be declared as `const` as it isn't a literal, so declare it as `private static readonly`.